libs/components/router/src/lib/modules/href/href.directive.ts

```diff
@@ -191,47 +189,28 @@ export class SkyHrefDirective {
   }
 
   #getChanges(): HrefChanges {
-    const queryParams: SkyHrefQueryParams = {};
-
     if (!this.#route || !this.#route.userHasAccess) {
       return {
         href: '',
         hidden: this.skyHrefElse === 'hide',
       };
     } else {
-      const [beforeFragment, fragment] = this.#route.url.split('#', 2);
-      const [baseUrl, search] = beforeFragment.split('?', 2);
-
-      if (search) {
-        const searchParams = new HttpParams({ fromString: search });
-        searchParams.keys().forEach((key) => {
-          queryParams[key] = searchParams.get(key);
-        });
-      }
+      const url = this.#route.url;
 
-      const queryParamsMerged = new HttpParams({
-        fromObject: Object.assign({}, this.#getSkyuxParams(), queryParams),
-      });
+      this.#href = this.#skyAppConfig?.runtime.params.getLinkUrl
```

Comment:
I don't like to see long chains of properties repeated because it's inefficient and doesn't minify well. Could you do something like this?

```
const params = this.#skyAppConfig?.runtime.params ?? this.#paramsProvider!.params;
this.#href = params.getLinkurl(url);
```

Reply:
Much better.